### ISSUE TITLE
Respecto aos imports

### ISSUE THREAD

**Issue by ssaavedra** [franrua/charla_1#6]
1. Import non carga tódalas funcións do módulof. Ou sí.
    1. Se hai un ficheiro __init__.py, este ficheiro regula (na súa variable __all__) qué cousas se cargan ao cargar o módulo ao que se refire a carpeta na que se atopa o ficheiro __init__.py.
    2. Se non hai tal ficheiro non recordo de memoria.
2. from x import y tamén carga todo (en esencia os dous chaman á función máxica `__import__()`) pero este último *importa* a(s) variable(s)/función(s) cargada(s) ao espazo de nomes raíz.

E digo *importa a variable ao espazo de nomes*, e non *importa o espazo de nomes* pois o que fai é meter a variable no espazo de nomes no que temos o resto de variables. En contraparte, `import x` podería dicirse que sí *importa o módulo `x` no espazo de nomes `x`*.

Perdoade o quisquilloso do asunto, pero tal e como estaba escrito parece querer dicir o contrario e igual alguén se confunde ao comparar coa documentación oficial ou outras fontes.

**Comment by franrua** [comment]
Modifiquei a explicación, pero aínda así non sei se queda totalmente claro. Espero que coas explicacións da clase quede claro. De todas formas nn sei se entrar en moito detalle nese asunto porque igual se fai un pouco complicado

**Comment by ssaavedra** [comment]
Sí, incluso podes contar menos. A explicación toda foi para falar eu con precisión, polo erro de importar x no espazo de nomes vs. importar o módulo como obxecto.

Ao final a conclusión é que `import x` fai que agora os contidos do módulo `x` estean dispoñibles *dentro* do obxecto nomeado agora `x` que representa o módulo; mentres que `from x import a, b, c` fai que `a`, `b` e `c` pasen a ser nomes das funcións respectivas do módulo `x`, do que ao resto de cousas que poida haber non se lles da visibilidade.

Pero sí, é entrar moito en detalle.
Perdoa que che conteste agora ^_^

Non lle deas máis importancia ;-)
